design/authenticators/authn_sut/authn_sut_solution_design.md

```diff
@@ -0,0 +1,310 @@
+a SUT for the user using their stored username and password or API key. PSM will
+then pass the SUT to the Conjur UI via a URL parameter. The Conjur UI will use
+the SUT to authenticate the user to Conjur and start a session.
+
+## Assumptions and Prerequisites
+
+Assumptions: Defining the assumptions being made in preparing this Solution
+Design. Detailing why these assumptions are valid and/or acceptable.
+Prerequisites: Do we need additional tooling to develop this feature? Do we need
+additional tooling to test this feature in our pipeline? What needs to be in
+place to build this feature?
+
+## Out-of-scope
+
+- SUTs will not be able to be used to authenticate to Conjur via the CLI
+  - SUTs are intended to be used in automated processes. The CLI is intended to
+    be used by humans (developers, operators, etc.)
+- There are many possible use cases for SUTs. This effort will focus on the
+  basic use case necessary for Conjur UI-PSM integration. Additional use cases
+  may be considered in the future.
+- The PSM integration itself will not be part of this effort. This effort will
+  only focus on the SUT authenticator and the API endpoint for generating SUTs.
+  The PSM team will implement the PSM integration in a separate effort in the
+  PSM codebase.
+
+## Design
+
+### High-Level Design
+
+#### Workflows
+
+##### Basic SUT Workflow
+
+The following diagram shows the basic workflow for using a SUT to authenticate
+to Conjur. The first step is to call the
+`GET /authn-sut/<account>/login` endpoint with the user's
+credentials to generate a SUT. The SUT is then passed to the
+`POST /authn-sut/<account>/authenticate`
+endpoint to authenticate the user to Conjur and retrieve a short-lived auth
+token.
+
+![Basic SUT Workflow](./sut.png)
+
+##### 3rd Party Client Workflow
+
+The following diagram shows the workflow for using a SUT to authenticate to
+Conjur on behalf of a 3rd party client. The first step is to call the
+`GET /authn-sut/<account>/login` endpoint with the user's
+credentials to generate a SUT. The SUT is then handed off to the 3rd party
+client. The 3rd party client then calls the
+`POST /authn-sut/<account>/authenticate`
+endpoint with the SUT to authenticate the user to Conjur and retrieve a
+short-lived auth token. This way the 3rd party client never has access to the
+user's credentials.
+
+![3rd Party Client Workflow](./sut_3rd_party.png)
+
+##### PSM Integration Workflow
+
+The following diagram shows the workflow for using a SUT to authenticate to
+Conjur via the PSM integration. The first step is to call the
+`GET /authn-sut/<account>/login` endpoint with the user's
+credentials to generate a SUT. The SUT is then passed to the Conjur UI via a URL
+parameter: `GET <conjur-ui-url>/ui/callback?sut=<sut>`. The Conjur UI then calls
+the `POST /authn-sut/<account>/authenticate` endpoint with
+the SUT to authenticate the user to Conjur and retrieve a short-lived auth
+token. The Conjur UI then starts a session for the user.
```

Comment:
Whats the default TTL of the auth token? Configurable or fixed? - TBD?
What happens once it expires? Does the user gets logged out of the UI?

Reply:
I am assuming this is regardless of the session activity? As long as PSM is okay with it, we should be good.

Reply:
Correct.